### PR TITLE
fix: add `get_all` param (and `--get-all`) to allow passing all to API

### PR DESCRIPTION
**Pull request by nejch**
When the `all=True` option was originally added, it probably wasn't anticipated that the GitLab API would have the same for parameters, making it unusable in some cases as the library consumes the argument (especially with the CLI). Closes https://github.com/python-gitlab/python-gitlab/issues/751.

This is a follow-up in trying to lower the confusion around `all` params and endpoints (getting rid of the `all()` method was the first step IMO). 

Although currently there is an ugly workaround for this (https://github.com/python-gitlab/python-gitlab/issues/751#issuecomment-698855273), I think it'd be best if we have more explicit arguments that don't clash with GitLab API params. At the same time I didn't want to change this completely as `all=True` is probably the most widely used argument for most list endpoints.

But I'd like to hear opinion on this approach  @max-wittig @JohnVillalovos :) I _guess_ gitlab could also introduce an `get_all` API param and then we're screwed again but I think that's unlikely.

Closes https://github.com/python-gitlab/python-gitlab/issues/751.
Related to https://github.com/python-gitlab/python-gitlab/issues/593.

### CODE REVIEW

**Review by JohnVillalovos — APPROVED** [review]
Thanks. This looks good to me.

**Comment by JohnVillalovos** [comment]
@nejch Any updates on this? I approved this back in May. Do you want to rebase and we can work on getting it merged.

**Comment by nejch** [comment]
> @nejch Any updates on this? I approved this back in May. Do you want to rebase and we can work on getting it merged.

I also received a comment on the initial issue that it might be confusing to have both, which was a good point. So I might replace this in all the docs as well, and just keep `all` for compatibility only.

**Comment by JohnVillalovos** [comment]
I sometimes wonder if the real long term fix for this type of issue is to only use `query_params` for extra parameters to send to GitLab. I realize it is convenient to just pass `**kwargs` around, but I'm not sure that is the best thing to do.

**Comment by nejch** [comment]
> I sometimes wonder if the real long term fix for this type of issue is to only use `query_params` for extra parameters to send to GitLab. I realize it is convenient to just pass `**kwargs` around, but I'm not sure that is the best thing to do.

I was thinking about this as well but it would really diminish the convenience of using this library vs. just using `requests` for example, which some people do. It would be a lot more verbose IMO.

File: docs/api-usage.rst [hunk 1]
```diff
@@ -205,11 +207,16 @@ listing methods support the ``page`` and ``per_page`` parameters:
...
-parameter to get all the items when using listing methods:
+parameter to get all the items when using listing methods. Alternatively, if
+the endpoint also accepts an ``all`` parameter itself, you can use the
+``all_pages`` parameter for listing methods to avoid this conflict:
 
 .. code-block:: python
 
    all_groups = gl.groups.list(all=True)
+   # or
+   all_groups = gl.groups.list(all_pages=True)
```
Comment (JohnVillalovos) [inline docs/api-usage.rst:218]: Curious. Can they also use:
```
gl.groups.list(all=True, query_parameters={"all": True})
```

Curious, but also wondering if we should document that, if it does work.

File: tests/functional/api/test_gitlab.py [hunk 2]
```diff
@@ -217,8 +217,8 @@ def test_list_all_false_nowarning(gl, recwarn):
 
 
 def test_list_all_true_nowarning(gl, recwarn):
-    """Using `all=True` will disable the warning"""
-    items = gl.gitlabciymls.list(all=True)
+    """Using `get_all=True` will disable the warning"""
+    items = gl.gitlabciymls.list(get_all=True)
     assert not recwarn
```
Comment (JohnVillalovos) [inline tests/functional/api/test_gitlab.py:222]: Can we also make sure to have a test with just `all=True` to make sure we aren't breaking anything?

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: tests/functional/api/test_gitlab.py (hunk 2 above)
Comment (nejch) [reply tests/functional/api/test_gitlab.py:233]: Might seem a little lazy but I parametrized the existing tests to test both in https://github.com/python-gitlab/python-gitlab/pull/1458/commits/6d69d3317894bd9a4b4c289b1ed9d8a87871ca1f in all our tests where we had `all=True` before.

Still not entirely sure how I feel about this PR even after over a year. But - it fixes things! :sweat_smile:

**Review by JohnVillalovos — APPROVED** [review]
Thanks @nejch

**Pull request merged**